File: sdks/ruby/lib/novabilling/types/batch_event_response.rb
# frozen_string_literal: true

module Novabilling
  module Types
    class BatchEventResponse < Internal::Types::Model
      field :received, -> { Integer }, optional: false, nullable: false
      field :processed, -> { Integer }, optional: false, nullable: false
      field :duplicates, -> { Integer }, optional: false, nullable: false
    end
  end
end

File: sdks/csharp/src/NovaBilling/Types/BatchEventResponse.cs
using System.Text.Json;
using System.Text.Json.Serialization;
using NovaBilling.Core;

namespace NovaBilling;

[Serializable]
public record BatchEventResponse : IJsonOnDeserialized
{
    [JsonExtensionData]
    private readonly IDictionary<string, JsonElement> _extensionData =
        new Dictionary<string, JsonElement>();

    [JsonPropertyName("received")]
    public required double Received { get; set; }

    [JsonPropertyName("processed")]
    public required double Processed { get; set; }

    [JsonPropertyName("duplicates")]
    public required double Duplicates { get; set; }

    [JsonIgnore]
    public ReadOnlyAdditionalProperties AdditionalProperties { get; private set; } = new();

    void IJsonOnDeserialized.OnDeserialized() =>
        AdditionalProperties.CopyFromExtensionData(_extensionData);

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}
